lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
Just to illustrate the problem:

`function createPrototypeChainCtor(depth, protoCtor)
{
    var ctor = function() {};
    if (protoCtor)
        ctor.prototype = new protoCtor();
    for (var i = 0; i < 100; i++)
    {
       ctor.prototype["prop_" + depth + "_" + i] = i;
    }
    if (depth == 0) { return ctor; }
    return createPrototypeChainCtor( depth - 1, ctor);
}

var ctor = createPrototypeChainCtor(5);

var obj = new ctor();
var d = new Date()

for (var i = 0; i < 10000; i++)
{
    for (var s in obj)
    {
    }
}


console.log(new Date() - d);
`

This takes 3.5x more time after your change